crates/aptos/src/account/multisig_account.rs

```diff
@@ -116,19 +124,96 @@ impl CliCommand<TransactionSummary> for CreateTransaction {
             .await
             .map(|inner| inner.into())
     }
 }
 
+/// Check entry function against on-chain transaction proposal payload.
+#[derive(Debug, Parser)]
+pub struct CheckTransaction {
+    #[clap(flatten)]
+    pub(crate) multisig_account_with_sequence_number: MultisigAccountWithSequenceNumber,
+    #[clap(flatten)]
+    pub(crate) txn_options: TransactionOptions,
+    #[clap(flatten)]
+    pub(crate) entry_function_args: EntryFunctionArguments,
+}
+
+#[async_trait]
+impl CliCommand<serde_json::Value> for CheckTransaction {
+    fn command_name(&self) -> &'static str {
+        "CheckTransactionMultisig"
+    }
+
+    async fn execute(self) -> CliTypedResult<serde_json::Value> {
+        // Get multisig transaction via view function.
+        let multisig_transaction = &self
+            .txn_options
+            .view(ViewRequest {
+                function: "0x1::multisig_account::get_transaction".parse()?,
+                type_arguments: vec![],
+                arguments: vec![
+                    serde_json::Value::String(String::from(
+                        &self
+                            .multisig_account_with_sequence_number
+                            .multisig_account
+                            .multisig_address,
+                    )),
+                    serde_json::Value::String(
+                        self.multisig_account_with_sequence_number
+                            .sequence_number
+                            .to_string(),
+                    ),
+                ],
+            })
+            .await?[0];
+        // Get expected multisig transaction payload hash bytes from provided entry function.
+        let expected_payload_hash_bytes =
+            HashValue::sha3_256_of(&to_bytes::<MultisigTransactionPayload>(
+                &self.entry_function_args.try_into()?,
+            )?)
+            .to_vec();
+        // Get actual multisig payload hash field stored on-chain, as bytes.
+        let mut actual_payload_hash_bytes =
+            view_json_option_hex_as_bytes(&multisig_transaction["payload_hash"])?;
+        // If payload hash not stored on-chain, get actual payload hash from on-chain payload.
+        if actual_payload_hash_bytes.is_empty() {
+            actual_payload_hash_bytes = HashValue::sha3_256_of(&view_json_option_hex_as_bytes(
+                &multisig_transaction["payload"],
+            )?)
+            .to_vec();
+        }
+        // If a match between expected payload hash and actual payload hash, return success message.
+        if expected_payload_hash_bytes.eq(&actual_payload_hash_bytes) {
+            Ok(json!({
+                "Status": "Transaction match",
+                "Multisig transaction": multisig_transaction
+            }))
+        // If a mismatch between expected payload hash and actual payload hash, error out.
+        } else {
+            Err(CliError::UnexpectedError(
+                "Transaction mismatch".to_string(),
```

Comment:
Is there anything more we can add here to demonstrate the mismatch? Or at the least guide users in the right direction to fix the problem. It would also be good to explain the implications of this mismatch. Even a link to the docs explaining what this means would be good.

Reply:
https://github.com/aptos-labs/aptos-core/blob/6aa74479f6b3f2b44f06cc0aea7efbd9acff1a51/crates/aptos/src/account/multisig_account.rs#L218-L222